bugsnag-android-core/src/main/java/com/bugsnag/android/ReportDeliveryDelegate.java

```diff
@@ -0,0 +1,109 @@
+        this.logger = logger;
+        this.eventStore = eventStore;
+        this.immutableConfig = immutableConfig;
+        this.breadcrumbState = breadcrumbState;
+    }
+
+    void deliverEvent(@NonNull Event event) {
+        // Build the report
+        Report report = new Report(immutableConfig.getApiKey(), event);
+        Session session = event.getSession();
+
+        if (session != null) {
+            setChanged();
+
+            if (event.isUnhandled()) {
+                event.setSession(session.incrementUnhandledAndCopy());
+                notifyObservers(StateEvent.NotifyUnhandled.INSTANCE);
+            } else {
+                event.setSession(session.incrementHandledAndCopy());
+                notifyObservers(StateEvent.NotifyHandled.INSTANCE);
+            }
+        }
+
+        if (event.isUnhandled()) {
+            eventStore.write(event);
+            eventStore.flushAsync();
+        } else {
+            deliverReportAsync(event, report);
+        }
+    }
+
+    private void deliverReportAsync(@NonNull Event event, Report report) {
+        final Report finalReport = report;
+        final Event finalEvent = event;
+
+        // Attempt to send the report in the background
+        try {
+            Async.run(new Runnable() {
+                @Override
+                public void run() {
+                    deliver(finalReport, finalEvent);
+                }
+            });
+        } catch (RejectedExecutionException exception) {
+            eventStore.write(event);
+            logger.w("Exceeded max queue count, saving to disk to send later");
+        }
+    }
+
+    DeliveryStatus deliver(@NonNull Report report, @NonNull Event event) {
+        DeliveryParams deliveryParams = immutableConfig.errorApiDeliveryParams();
+        Delivery delivery = immutableConfig.getDelivery();
+        DeliveryStatus deliveryStatus = delivery.deliver(report, deliveryParams);
+
+        switch (deliveryStatus) {
+            case DELIVERED:
+                logger.i("Sent 1 new event to Bugsnag");
+                leaveErrorBreadcrumb(event);
+                break;
+            case UNDELIVERED:
+                logger.w("Could not send event(s) to Bugsnag,"
+                        + " saving to disk to send later");
+                eventStore.write(event);
+                leaveErrorBreadcrumb(event);
+                break;
+            case FAILURE:
+                logger.w("Problem sending event to Bugsnag");
+                break;
+            default:
+                break;
+        }
+        return deliveryStatus;
+    }
+
+    private void leaveErrorBreadcrumb(@NonNull Event event) {
+        // Add a breadcrumb for this event occurring
+        List<Error> errors = event.getErrors();
+
+        String name = "";
+        String msg = "";
+
+        if (errors.size() > 0) {
+            name = errors.get(0).getErrorClass();
+            msg = errors.get(0).getErrorMessage();
+        }
```

Comment:
This was discussed and refactored here: https://github.com/bugsnag/bugsnag-android/pull/673/files#diff-d01d476975cb3e9ebc74fb3cec71c943R759-R764

Is this a deliberate change or a regression?

Reply:
This was a regression due to copying over from the old diff, I've updated to use the refactored version